homeassistant/components/light/flux_led.py

```diff
@@ -253,25 +248,32 @@ def turn_on(self, **kwargs):
         if brightness is None:
             brightness = self.brightness
 
+        if hs_color and brightness:
+            self._color = color_util.color_hsv_to_RGB(hs_color[0], hs_color[1],
+                                                      brightness / 255 * 100)
+
         # Preserve color on brightness/white level change
         if rgb is None:
-            rgb = self._bulb.getRgb()
-
-        if white is None and self._mode == MODE_RGBW:
-            white = self.white_value
+            rgb = self._color
 
         # handle W only mode (use brightness instead of white value)
         if self._mode == MODE_WHITE:
             self._bulb.setRgbw(0, 0, 0, w=brightness)
 
         # handle RGBW mode
         elif self._mode == MODE_RGBW:
-            self._bulb.setRgbw(*tuple(rgb), w=white, brightness=brightness)
+            if white is None:
+                self._bulb.setRgbw(*tuple(rgb))
+            else:
+                self._bulb.setRgbw(w=white)
 
         # handle RGB mode
         else:
             self._bulb.setRgb(*tuple(rgb), brightness=brightness)
 
+        from time import sleep
+        sleep(2)
```

Comment:
It is not allowed to use `time.sleep()` since this will block other integrations from using the thread.

Reply:
Oh, I just used it for testing and forgot to remove it, thank you!

Reply:
Some of the models seem to have a transition period ≈1 second, there needs to be a delay or scheduled call for those models.

Reply:
If you need to sleep, the best way is probably to implement `async_turn_on()` and use `asyncio.sleep()`

Let me know if you need help with that.

Reply:
So, it's now with asyncio sleep.